react-hot-button-component/src/hot-button.jsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-export class HotButton extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { clickCount: 0 };
+const HotButton = () => {
+  const [clickCount, setCount] = useState(() => {
+    return 0;
+  });
```

Comment:
If you are starting with a simple state like a primitive (number, string, null, etc), it's best to just stick with the literal initialization like this:

```js
const [clickCount, setCount] = useState(0);
```

The code is more concise and uses less memory than a function.